tests/level2_networking_test.go

```diff
@@ -5,193 +5,224 @@ package caf_tests
 import (
 	"context"
 	"fmt"
+	"strings"
 	"testing"
 
 	"github.com/gruntwork-io/terratest/modules/azure"
 	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
 )
 
 func TestThereAreTwoResourceGroupsForNetworkingHub(t *testing.T) {
 	t.Parallel()
 
-	test := prepareTestTable()
+	tfState := NewTerraformState(t, "networking_hub")
+	resourceGroups := tfState.GetResourceGroups()
+	client, _ := azure.GetResourceGroupClientE(tfState.SubscriptionID)
 
-	client, _ := azure.GetResourceGroupClientE(test.SubscriptionID)
+	for _, resourceGroup := range resourceGroups {
+		name := resourceGroup.GetName()
 
-	result, _ := client.List(context.Background(), "tagName eq 'level' and tagValue eq 'level2'", nil)
+		actual_rc, err := client.Get(context.Background(), name)
+		require.NoError(t, err)
 
-	rgList := result.Values()
-
-	actual := 0
-	for _, rg := range rgList {
-		if *rg.Tags["landingzone"] == "networking_hub" && *rg.Tags["environment"] == test.Environment {
-			actual++
-		}
+		assert.Equal(t, *actual_rc.Tags["landingzone"], tfState.GetLandingZoneKey())
+		assert.Equal(t, *actual_rc.Tags["environment"], tfState.Environment)
+		assert.Equal(t, *actual_rc.Tags["level"], resourceGroup.GetLevel())
 	}
-
-	expected := 2
-
-	assert.Equal(t, expected, actual, fmt.Sprintf("There must be %d resource group with 'landingzone=networking_hub' and 'environment=%s' tags, found %d", expected, test.Environment, actual))
 }
 
 func TestVirtualNetworksAreInDifferentRegions(t *testing.T) {
 	t.Parallel()
+	tfState := NewTerraformState(t, "networking_hub")
+	vNets := tfState.GetVNets()
 
-	test := prepareTestTable()
-
-	vnet1, _ := azure.GetVirtualNetworkE(fmt.Sprintf("%s-vnet-hub-re1", test.Prefix), fmt.Sprintf("%s-rg-vnet-hub-re1", test.Prefix), test.SubscriptionID)
+	var locations []string
+	for _, vnet := range vNets {
+		vn, err := azure.GetVirtualNetworkE(vnet.GetName(), vnet.GetString("resource_group_name"), tfState.SubscriptionID)
+		require.NoError(t, err)
 
-	vnet2, _ := azure.GetVirtualNetworkE(fmt.Sprintf("%s-vnet-hub-re2", test.Prefix), fmt.Sprintf("%s-rg-vnet-hub-re2", test.Prefix), test.SubscriptionID)
+		locations = append(locations, *vn.Location)
+	}
 
-	assert.NotEqual(t, *vnet1.Location, *vnet2.Location, "Virtual Networks in the 'landingzone=networking_hub' resource groups should provisioned in different regions")
+	assert.NotEqual(t, locations[0], locations[0], fmt.Sprintf("Virtual Networks in the 'landingzone=%s' resource groups should provisioned in different regions", tfState.GetLandingZoneKey()))
```

Comment:
Shouldn't this be testing locations[0] != locations[1] to ensure that the locations are not the same?

Reply:
fixed